examples/all-clusters-app/all-clusters-common/src/water-heater-mode.cpp

```diff
@@ -0,0 +1,105 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+#include <app-common/zap-generated/attributes/Accessors.h>
+#include <water-heater-mode.h>
+
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::WaterHeaterMode;
+using chip::Protocols::InteractionModel::Status;
+template <typename T>
+using List              = chip::app::DataModel::List<T>;
+using ModeTagStructType = chip::app::Clusters::detail::Structs::ModeTagStruct::Type;
+
+static ExampleWaterHeaterModeDelegate * gWaterHeaterModeDelegate = nullptr;
+static ModeBase::Instance * gWaterHeaterModeInstance             = nullptr;
+
+CHIP_ERROR ExampleWaterHeaterModeDelegate::Init()
+{
+    return CHIP_NO_ERROR;
+}
+
+// todo refactor code by making a parent class for all ModeInstance classes to reduce flash usage.
+void ExampleWaterHeaterModeDelegate::HandleChangeToMode(uint8_t NewMode, ModeBase::Commands::ChangeToModeResponse::Type & response)
+{
+    response.status = to_underlying(ModeBase::StatusCode::kGenericFailure);
```

Comment:
Please document why this is just refusing all mode changes?

Reply:
This should have returned (now does) ModeBase::StatusCode::kSuccess.

Reply:
Change made.